cpp/src/strings/convert/convert_fixed_point.cu

```diff
@@ -0,0 +1,450 @@
+  return type_dispatcher(output_type, dispatch_to_fixed_point_fn{}, input, output_type, stream, mr);
+}
+
+}  // namespace detail
+
+// external API
+std::unique_ptr<column> to_fixed_point(strings_column_view const& strings,
+                                       data_type output_type,
+                                       rmm::mr::device_memory_resource* mr)
+{
+  CUDF_FUNC_RANGE();
+  return detail::to_fixed_point(strings, output_type, rmm::cuda_stream_default, mr);
+}
+
+namespace detail {
+namespace {
+/**
+ * @brief Calculate the size of the each string required for
+ * converting each value in base-10 format.
+ *
+ * ouput format is [-]integer.fraction
+ */
+template <typename DecimalType>
+struct decimal_to_string_size_fn {
+  column_device_view const d_column;
+
+  __device__ int32_t operator()(size_type idx) const
+  {
+    if (d_column.is_null(idx)) return 0;
+    auto const value = d_column.element<DecimalType>(idx);
+    auto const scale = d_column.type().scale();
+
+    if (scale >= 0) return count_digits(value) + scale;
+
+    auto const abs_value = std::abs(value);
+    auto const exp_ten   = static_cast<int32_t>(exp10(static_cast<double>(-scale)));
+    auto const num_zeros = std::max(0, (-scale - count_digits(abs_value % exp_ten)));
+    return static_cast<int32_t>(value < 0) +    // sign if negative
+           count_digits(abs_value / exp_ten) +  // integer
+           1 +                                  // decimal point
+           num_zeros +                          // zeros padding
+           count_digits(abs_value % exp_ten);   // fraction
+  }
+};
+
+/**
+ * @brief Convert each value into a string.
+ *
+ * The value is converted into base-10 digits [0-9]
+ * plus the decimal point and a negative sign prefix.
+ */
+template <typename DecimalType>
+struct decimal_to_string_fn {
+  column_device_view const d_column;
+  int32_t const* d_offsets;
+  char* d_chars;
+
+  __device__ void operator()(size_type idx)
+  {
+    if (d_column.is_null(idx)) return;
+    auto const value = d_column.element<DecimalType>(idx);
+    auto const scale = d_column.type().scale();
+    char* d_buffer   = d_chars + d_offsets[idx];
+
+    if (scale >= 0) {
+      integer_to_string(value, d_buffer);
+      d_buffer += count_digits(value);
+      thrust::generate_n(thrust::seq, d_buffer, scale, []() { return '0'; });  // add zeros
+      return;
+    }
+
+    // scale < 0
+    // write format:   [-]integer.fraction
+    // where integer  = abs(value) / (10^abs(scale))
+    //       fraction = abs(value) % (10^abs(scale))
+    auto const abs_value = std::abs(value);
+    if (value < 0) *d_buffer++ = '-';  // add sign
+    auto const exp_ten   = static_cast<int32_t>(exp10(static_cast<double>(-scale)));
+    auto const num_zeros = std::max(0, (-scale - count_digits(abs_value % exp_ten)));
+
+    integer_to_string(abs_value / exp_ten, d_buffer);  // add the integer part
+    d_buffer += count_digits(abs_value / exp_ten);
```

Comment:
If `integers_to_string` had the count, then I would simply return it but since it does not, I would have to add logic to compute and return it. Since this is a utility that is used by other functions, I did not want to add this cost to those functions that do not need the count.

Reply:
This turned out to be only one line of code plus the return. I think this is a good idea.